src/components/NewClusterModal.js

```diff
@@ -139,24 +136,35 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
   constructor(props) {
     super(props)
     const currentCluster = this.getCurrentCluster()
```

Comment:
I notice that `this.getCurrentCluster()` and `this.getCurrentPersistentDisk()` are used for exactly 2 things:

1. Initializing the state backing the form elements (`constructor`)
2. Grabbing the keys used to fetch the details (`componentDidMount`)
(Yes, there is also a 3rd use of logging the event of opening the panel, but that is also done in `componentDidMount`.)

Is there a pattern we can use to make this more clear? Inlining the logic would make it more clear that the initialization is based on `props` while further operation of the form is based on fetched state, but would end up duplicating the code. Having a component-level function addresses the code duplication, but results in a function that reads like it could be used more broadly rather than just during initialization.

I just want to make sure that the dataflow is clear to future readers of this code.

Reply:
Yeah, I think we should write a comment that describes the high-level data flow in the modal, to help people understand how the pieces fit together.